projects/plugins/jetpack/modules/masterbar/admin-menu/class-wpcom-admin-menu.php

```diff
@@ -250,18 +250,42 @@ public function add_stats_menu() {
 		add_menu_page( __( 'Stats', 'jetpack' ), $menu_title, 'edit_posts', 'https://wordpress.com/stats/day/' . $this->domain, null, 'dashicons-chart-bar', 3 );
 	}
 
+	/**
+	 * Holds the current product, set by get_current_product().
+	 *
+	 * @var array
+	 */
+	private $cached_product = array();
```

Comment:
Let's move that property to the top of the class definition. Would `$current_plan` be a bit more granular than cached_product?

Reply:
> Let's move that property to the top of the class definition.

👍 

> Would $current_plan be a bit more granular than cached_product?

It would. I started that route at first, but opted to store the whole [product array](https://user-images.githubusercontent.com/77539/146054790-848daf2c-21fa-4e29-9aa3-ce7524925508.png), because while the add_upgrades_menu() is interested in the `product_name_short` value, we're using the `product_id` for comparing plans in the filter: D69877-code.

Reply:
`$plan_product`? Cached product just feels rather ambiguous to me

Reply:
Oh, I see. I thought you meant storing a single value over the whole array. My misunderstanding. 😅

Yeah, I actually like `$current_plan` or `$plan_product`. I'm not really fond of `$cached_product`. Because "cached" could be misconstrued. But my fear was someone might reference `$current_product` inside this class and not realize they should instead `get_current_product()`. In which case they might assume a `null` value means the site has no plan. That was my thinking at least. But, that variable is private and there is also documentation in the class. -- I'll change it to `$current_plan`.

Reply:
Renamed the class variable to $current_plan and took it a step further and refactored our use of "product" to "plan" for consistency. It's `get_current_plan()` now for example.